### PR TITLE
[Fluid] Clean up & update schemes

### PR DESCRIPTION
**Pull request by rubenzorrilla**
This updates the `FluidDynamicsApplication` schemes according to the API changes in the base class.

I also took the chance to do some clean up. This includes creating a `schemes` folder (so far the schemes were stored in the `strategies` folder) and to remove the two legacy BDF schemes, which are no longer used.

### CODE REVIEW

File: applications/FluidDynamicsApplication/custom_strategies/schemes/residualbased_predictorcorrector_velocity_bossak_scheme_turbulent.h [hunk 1]
```diff
@@ -697,7 +694,7 @@ namespace Kratos {
 
                 //adding the dynamic contributions (statics is already included)
                 AddDynamicsToLHS(LHS_Contribution, mDamp[thread_id], mMass[thread_id], CurrentProcessInfo);
-                AddDynamicsToRHS((*itElem), RHS_Contribution, mDamp[thread_id], mMass[thread_id], CurrentProcessInfo);
+                AddDynamicsToRHS(*(*itElem), RHS_Contribution, mDamp[thread_id], mMass[thread_id], CurrentProcessInfo);
```
Comment (philbucher) [inline applications/FluidDynamicsApplication/custom_strategies/schemes/residualbased_predictorcorrector_velocity_bossak_scheme_turbulent.h]:
```suggestion
                AddDynamicsToRHS(**itElem, RHS_Contribution, mDamp[thread_id], mMass[thread_id], CurrentProcessInfo);
```

**Review by philbucher — DISMISSED** [review]
awesome, thx!

**Comment by philbucher** [comment]
@sunethwarna probably good if you can run one of your cases with this, just to make sure :)
Thx

**Review by philbucher — APPROVED** [review]

**Comment by ddiezrod** [comment]
We use the residualbased_predictorcorrector_velocity_bdf_scheme_turbulent_no_reaction, is there any specific reason why it was removed? @philbucher @rubenzorrilla

**Comment by rubenzorrilla** [comment]
> We use the residualbased_predictorcorrector_velocity_bdf_scheme_turbulent_no_reaction, is there any specific reason why it was removed? @philbucher @rubenzorrilla

We found no usages throughout the entire repo. The "official" way to deal with elements that internally manage the integration is the `ResidualBasedIncrementalUpdateStaticSchemeSlip`. Were you using the legacy one because you require a turbulence model?

If so it is, I think we should create a derived class from the one in the core.

**Comment by ddiezrod** [comment]
Ok, @rubenzorrilla , I will try to use that one then. Thank you! Just for notification @KratosMultiphysics/altair

**Comment by rubenzorrilla** [comment]
> Ok, @rubenzorrilla , I will try to use that one then. Thank you! Just for notification @KratosMultiphysics/altair

Just note that the `ResidualBasedIncrementalUpdateStaticSchemeSlip` only takes care of the Dx update. Hence, you need to take care of the calculation of the BDF coefficients vector (we use the `TimeDiscretization` utility for that).

**Comment by ddiezrod** [comment]
Yes, at first sight it is very similar to the old scheme in FluDyn. One question, is there an MPI version ResidualBasedIncrementalUpdateStaticSchemeSlip ? @rubenzorrilla

**Comment by rubenzorrilla** [comment]
> Yes, at first sight it is very similar to the old scheme in FluDyn. One question, is there an MPI version ResidualBasedIncrementalUpdateStaticSchemeSlip ? @rubenzorrilla

There is. The `TrilinosResidualBasedIncrementalUpdateStaticSchemeSlip`.

**Comment by ddiezrod** [comment]
I see, I was trying to find a different file but it is the very same one. Thanks a lot @rubenzorrilla !

**Comment by rubenzorrilla** [comment]
> I see, I was trying to find a different file but it is the very same one. Thanks a lot @rubenzorrilla !

:wink:

**Comment by ddiezrod** [comment]
The deprecated ResidualBasedPredictorCorrectorBDFSchemeTurbulent had a Predict method added by @RiccardoRossi. It basically did `vn+1 = vn + dt*(vn - vn-1)/oldDt`. Do you think we should keep this method? We can override the scheme in our side anyway if you prefer so.

**Comment by ddiezrod** [comment]
@pablobecker says the Predict method was not working fine in their cases, so I guess there is no reason why we should put in the base class.

**Pull request merged**